File: ApiHeroku/Data/Model/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ApiHeroku.Data.Model
{
    public class Booking
    {
        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }

        [property: JsonConverter(typeof(DateOnlyConverter))]
        public DateOnly from_date { get; set; }

        [property: JsonConverter(typeof(DateOnlyConverter))]
        public DateOnly to_date { get; set; }

        [property: JsonConverter(typeof(DateOnlyConverter))]
        public DateOnly booking_date { get; set; }
        
        //many-to-many relationship
        public int RoomId { get; set; }
        public Room Room { get; set; }
        public int UserId { get; set; }
        public UserExample UserExample { get; set; }
    }
}

File: BookingMaster/app/src/main/java/com/example/bookingmaster/model/Booking.kt
package com.example.bookingmaster.model

import com.squareup.moshi.JsonClass

/** Post booking **/
@JsonClass(generateAdapter = true)
data class BookingResponse(val booking : Booking, val message : String, val code : Int)

@JsonClass(generateAdapter = true)
data class BookingRequest(
    val from_date: String,
    var to_date: String,
    var RoomId: Int,
    val UserId: Int,
)

@JsonClass(generateAdapter = true)
data class Booking(
    var booking_date: String
)